### ISSUE TITLE
Sporadic warnings in log referencing invalid input_selection options.

### ISSUE THREAD

**Issue by breakthestatic** [home-assistant/frontend#11991]
### The problem

I'm getting sporadic warnings in the logs regarding a service call to input_select.select_option with an invalid option (in this case *no* option.  I've confirmed that **all** my automation/script calls are properly passing valid data, and am also **not** seeing the issue anywhere around the time that my automations run.

It seems to pop up when I navigate to my main lovelace dashboard after a period of inactivity. The dashboard has an `entities` card containing the input_select in question.

### What version of Home Assistant Core has the issue?

core-2022.3.3

### What was the last working version of Home Assistant Core?

core-2022.2

### What type of installation are you running?

Home Assistant Container

### Integration causing the issue

input_select

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/input_select/

### Anything in the logs that might be useful for us?

```txt
2022-03-08 12:25:01 DEBUG (MainThread) [homeassistant.components.websocket_api.http.connection] [140629127355216] Received {'type': 'call_service', 'domain': 'input_select', 'service': 'select_option', 'service_data': {'option': '', 'entity_id': 'input_select.home_monitor'}, 'id': 27}
2022-03-08 12:25:01 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=input_select, service=select_option, service_data=option=, entity_id=input_select.home_monitor>
2022-03-08 12:25:01 WARNING (MainThread) [homeassistant.components.input_select] Invalid option:  (possible options: Home, Away, Sleep, Guest)
```

**Comment by balloob** [comment]
We've changed the dropdown component and this is probably an unintentional side effect. I've seen similar in the past.

**Comment by zsarnett** [comment]
Looking into this but was unable to reproduce it.

@Gferretta 

> Lovelace' displaying was empty for that input_select helpers when this error occurs

Is the value of the input blank for does the input_select have a value but its shown as blank or '' in the frontend?

**Comment by Gferretta** [comment]
> Looking into this but was unable to reproduce it.
> 
> @Gferretta
> 
> > Lovelace' displaying was empty for that input_select helpers when this error occurs
> 
> Is the value of the input blank for does the input_select have a value but its shown as blank or '' in the frontend?

I forgot to mention that the "no value" visual problem I noticed for now only in the android version of HA.
But the error also appears in the PC version, only that the input_select does not appear "without value"

input_select has values but in the Android frontend it doesn't show anything, not even the default value.

Thanks

**Comment by Gferretta** [comment]
NEWS

This happen now in PC Chrome 
Lovelace don't show value of input_text


![HA 1](https://user-images.githubusercontent.com/65413059/157750030-13625ba3-b81b-4e00-a578-b914c6dd97dd.jpg)


![HA 2](https://user-images.githubusercontent.com/65413059/157750053-2ffd25ba-b354-4335-9e72-bdb9c2b057b5.jpg)


did this help you?

**Comment by zsarnett** [comment]
What is the current state of that input_select at the times it does that?

**Comment by Gferretta** [comment]
> What is the current state of that input_select at the times it does that?
> 
> (you can find that via developer tools)

Default value (1st)

**Comment by Gferretta** [comment]
Idem error but in Android app

![Screenshot_20220310-233307_Home Assistant.jpg](https://user-images.githubusercontent.com/65413059/157795149-e0e2f667-87ad-4b41-a863-40b6e193d37a.jpg)


![Screenshot_20220310-233337_Home Assistant.jpg](https://user-images.githubusercontent.com/65413059/157795169-18227f87-3c4b-43e6-aa00-cd1aed0b16d1.jpg)


 ![Screenshot_20220310-233434_Home Assistant.jpg](https://user-images.githubusercontent.com/65413059/157795183-7725d5e1-2b5c-45c6-b890-b4d1ad9c885b.jpg)

**Comment by balloob** [comment]
@zsarnett the empty input is the same as with translations -> the list items have not rendered when the select is mounted and so it cannot read the text of the selected option.

That is however not what this issue is about. The frontend can sometimes trigger value changed events and causing the set_option service to be called.

**Comment by zsarnett** [comment]
We guard against it if the value is the state of the entity. That's why I am wondering why it's sending an empty string to the service.. 

I know we can sometimes cause it to run before it is ready but it shouldn't be an empty string I would think

**Comment by balloob** [comment]
We don't guard for it. We only guard it's not the same value https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts#L97

**Comment by balloob** [comment]
The problem is that it is sending an empty value, which is not a valid option. I wonder if it can happen if `selected` fires before children are rendered ?

**Comment by zsarnett** [comment]
Yea thats what I was thinking as well... First I was wondering if possible we are trying to set the state of the entity if no state exist as '' but I think that is possible that we fire before the children get there. so I may try throttling my connection and try to reproduce

**Comment by balloob** [comment]
We could just check the data we're about to be sent is actually part of the allowed options stored in the attributes.

**Comment by lmagyar** [comment]
Some additional info (noticed it with Chrome, under Win, everything is latest version):
- it usually happens after long inactivity (opening several other pages, maybe Chrome empties some cache, and when I switch back to the HA page with entities/input_select in it, it redraws itself and the empty state update happens around this time)
- after this, a few seconds later, switching to other HA tabs, the UI freezes, CPU 100%, Chrome thread crashes or I kill it manually and then refresh and everything is fine for some time (usually for hours)

And this is "random", really unpredictable when it happens, there is no deterministic way to reproduce this. But it happens several times a day in the past few weeks.

I'm not a web developer, but if you need some kind of log from Chrome/HA UI, tell me what to do to get these logs/information. After a day, I'm sure I will run into this problem again.

**Comment by lmagyar** [comment]
I'm still getting the same errors:
```
2022-04-08 08:40:38 WARNING (MainThread) [homeassistant.components.input_select] Invalid option:  (possible options: Default, Preheat)
```

Frontend version: 20220405.0 (that contains this fix)

So the root cause seems to somewhere else, I think this is issue should be reopened.

**Comment by zsarnett** [comment]
Did you see this in the logs before loading any other UI pages? I guess your other devices could be actively showing the UI which could cause it

**Comment by breakthestatic** [comment]
For the first instance (immediately after the upgrade), I first opened the root url (effectively loading my default dashboard which **does** have an input_select on it). From there, I went immediately to the logs to see the database upgrade progress, which is where I also noticed the "invalid option" error again.

After that, the subsequent instances, I still haven't *completely* pinpointed it, but it seems to be highly correlated to coming back to HA after some period of inactivity.  Once it appears in the logs, no amount of immediate navigation (trying to induce it) seems to do anything.

**Comment by balloob** [comment]
What components do you use to render your input select? Just the default entity row ?

**Comment by breakthestatic** [comment]
Yup, it's just an entities card with the input_select as the single entity:
```
type: entities
entities:
  - input_select.home_monitor

```

**Comment by lmagyar** [comment]
> Did you see this in the logs before loading any other UI pages? I guess your other devices could be actively showing the UI which could cause it

For 6 days my laptop (Win+Chrome) was shut down, I used only my phone (Android+App), only a few times a day to access the UI, and there were no errors for 5 days, but then on the 6th day a new `Invalid option:` error again. I'm 100% sure, that only my phone was using the UI at that time alone.

**Comment by balloob** [comment]
Please make sure that you are for sure running the latest frontend. If you're using an old version of the frontend, you are not running my fix.

**Comment by cdonovan** [comment]
I'm still seeing this on HA 2022.4.4 and Frontend version: 20220405.0

**Comment by breakthestatic** [comment]
I'm still seeing this on HA 2022.4.4 and Frontend version: 20220405.0

**Comment by hellcry37** [comment]
also related to this: https://github.com/home-assistant/core/issues/69178

**Comment by Mariusthvdb** [comment]
And maybe this? https://github.com/home-assistant/core/issues/65669

**Comment by breakthestatic** [comment]
Hey @balloob, I was just revisiting this issue and realized that your fix in #12181 is for hui-select-entity-row.ts, However, my issue involves an **input_select** entity, which I'm guessing is actually rendered by hui-input-select-entity-row.ts (and which doesn't have the extra guard):
https://github.com/home-assistant/frontend/blob/0bbc0ebb3c92ed66e741388d164c0d34085f939e/src/panels/lovelace/entity-rows/hui-input-select-entity-row.ts#L94-L104

**Comment by zsarnett** [comment]
Good Eye. Ill add that
